### PR TITLE
[IMP] mail: chat window in mobile public page

### PR DESCRIPTION
**Pull request by Atovange**
With this commit the mobile view of the public page is a fullscreen chat window just like regular discuss.

Before:

After:

task-3346088
task-2664822

### CODE REVIEW

File: addons/mail/static/src/core/common/chat_window.xml [hunk 1]
```diff
@@ -67,6 +67,18 @@
...
+                    <img class="o-mail-Discuss-selfAvatar rounded-circle o_object_fit_cover flex-shrink-0" alt="Avatar" t-att-src="store.self.avatarUrl" t-on-click="() => state.editingGuestName = true"/>
+                </button>
+                <AutoresizeInput
+                    t-if="state.editingGuestName"
+                    className="'py-1 me-2'"
+                    autofocus="true"
+                    onValidate.bind="renameGuest"
+                    value="store.self.name"
+                />
+            </t>
```
Comment (alexkuhn) [inline addons/mail/static/src/core/common/chat_window.xml:81]: Only guests can change their name. Internal users cannot (this is linked to their account / partner)

File: addons/mail/static/src/core/public_web/discuss.js [hunk 2]
```diff
@@ -67,6 +68,22 @@ export class Discuss extends Component {
             },
             { capture: true }
         );
+        if (this.store.inPublicPage) {
+            threadActionsRegistry.remove("close");
```
Comment (alexkuhn) [inline addons/mail/static/src/core/public_web/discuss.js:72]: Recently guests can have discuss sidebar:
https://github.com/odoo/odoo/pull/171016

So the close button should be disabled only if there's "no sidebar". The close button is important to go back to conversation selection in mobile!

File: addons/mail/static/src/core/common/chat_window.xml (hunk 1 above)
Comment (Atovange) [reply addons/mail/static/src/core/common/chat_window.xml:81]: The guard `this.store.inPublicPage` should already make sure this is only displayed to guests, does it not?

Comment (alexkuhn) [reply addons/mail/static/src/core/common/chat_window.xml:81]: No, you can access the public guest page while being authenticated with an internal user. So `this.store.inPublicPage` is not enough to ensure this is guest-only branching

**Review by alexkuhn — COMMENTED** [review]
Welcome page is not responsive in mobile. Because of this, it gives the impression we cannot pass the welcome page, although horizontal scrolling works.

Since this PR is about improving main UX/UI of mobile of discuss app, making the welcome page responsive seems important now

File: addons/mail/static/src/core/public_web/messaging_menu.js [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+                id: "chat",
+                label: _t("Chat"),
+            },
+            {
+                icon: "fa fa-users",
+                id: "channel",
+                label: _t("Channel"),
+            },
+        ];
+    }
```
Comment (alexkuhn) [inline addons/mail/static/src/core/public_web/messaging_menu.js:139]: `"livechat"` and `"whatsapp"` channels can also be access in public page. These patches should also be in `public` bundle

**Review by alexkuhn — COMMENTED** [review]
LGTM otherwise :)

**Review by alexkuhn — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**